### DISCUSSION TITLE
Trying to build a development project launcher workflow

### DISCUSSION THREAD

**Discussion by mearleycf** [ttscoff/bunch-docs#331] [Q&A]
So, I've got a series of say 5 or 6 development projects I'm working on. In each instance, I'm doing some combination of the following:
- open kitty to a specific application directory
```
project_name = ?[balanceit, fillit, manageit, backit, astro-base, portfolio-2024] "Which project?"

%kitty
- ~/code/${project_name}
```
That works. That's about the only thing I can get to work though. 
Basically, my workflow would be something like:
1. Open kitty to specifically selected project
2. Move kitty into the correct window position by executing a bettertouchtool named trigger url handler
3. Open visual studio code by opening the specifically selected project's folder; typically I would do this by navigating to that project folder in kitty, then running `code .`. Or, I can create a workspace file, and then theoretically this would work...see below for what actually happens. 
4. Open chatGPT and move to correct window location
5. Open specific URLs in Arc browser--so, if my project is the 'balanceit' project, for example, I want to open documentation about Plaid, my Plaid dashboard, the plaid-python github repo, and maybe 1 or 2 other sites. I don't even mind setting up a JSON file or other file type that Bunch can iterate over to open the appropriate windows for each project. Something like an object with key project_name and value of the project's name, then a key of "URLs" and an array of URL strings that the app should open...again, I just have NO clue how to do this. And I've been screwing around with it for hours. 
```
Visual Studio Code
- ~/code/${project_name}/${project_name}.code-workspace
```
But--it doesn't. Instead I get this:
```
2024-07-13 11:01:05: ⚪[dev] Open >>>>>>>>>>>>>>>>>>>>>
2024-07-13 11:01:05: ⚪Presenting multiple choice from array: (
    balanceit,
    fillit,
    manageit,
    backit,
    "astro-base",
    "portfolio-2024"
)
2024-07-13 11:01:06: ⚪Multiple Choice answered (balanceit)
2024-07-13 11:01:06: ⚪[dev] Opening file /Users/mikeearley/code/balanceit with app kitty
2024-07-13 11:01:06: ⚪[dev] Opening file /Users/mikeearley/code/balanceit/balanceit.code-workspace with app Visual Studio Code
2024-07-13 11:01:06: 🔴Application failed to launch (Code) and/or open file (/Users/mikeearley/code/balanceit/balanceit.code-workspace)
```
So I haven't even gotten very far. For example, after I open the workspace, I want to move that window to the appropriate window space as well, again in theory I can execute a btt URL handler to do that, but it doesn't work.

**Comment by ttscoff** [comment]
I’m speaking at Macstock this weekend so won’t be able to help much until next week, but I would love to help you sort this out, super useful use case. I imagine solving this will take some bug fixes in Bunch.

Thanks,
Brett

**Comment by ttscoff** [comment]
Ok, I'm back. So is the problem just the VS Code failing to launch? That may be related to another report I've received that the VS Code workaround is no longer functioning. I should have time to look at that later this week.

As far as the JSON file of related URLs, etc., it should be possible to populate variables differently based on a key (from the multiple choice), but it would require external scripting to output different JSON based on that key. There's no filtering available in Bunch (though it may be a worthy idea for me to pursue). In standard Bunch parlance, you would create [embedded snippets](https://bunchapp.co/docs/bunch-files/snippets/#embeddedsnippets), one for each menu item, and then call the snippets based on a logic fork (or name them appropriately and just use `<<#${project_name}`, which should work).

**Comment by ttscoff** [comment]
As a workaround for the VS Code stuff, if you have the `code` CLI installed, you should be able to just do:

```
$ code ~/code/${project_name}/${project_name}.code-workspace
```

Instead of the whole "Visual Studio Code" bit.

**Reply by ttscoff** [reply]
That won't quit VS Code when closing the Bunch, though.
